pkg/bicep/schematobicep.go

```diff
@@ -61,3 +64,196 @@ func getBicepType(schemaType string) (string, error) {
 		return "", errors.New("unknown type: " + schemaType)
 	}
 }
+
+func declareParameter(name string, sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	if sch.Default != nil {
+		err = declareDefault(name, sch, buf)
+		if err != nil {
+			return err
+		}
+		return nil
+	} else {
+		buf.WriteString(fmt.Sprintf("param %s %s\n", name, bicepType))
+	}
+
+	return nil
+}
+
+func declareAllowed(sch *schema.Schema, buf *bytes.Buffer) error {
+	if sch.Enum != nil && len(sch.Enum) > 0 {
+		bicepType, err := getBicepType(sch.Type)
+		if err != nil {
+			return err
+		}
+
+		enumBytes, err := json.MarshalIndent(sch.Enum, "", "    ")
+		if err != nil {
+			return err
+		}
+
+		enumString := string(enumBytes)
+		r := strings.NewReplacer(`"`, `'`, ",", "")
+		cleanString := r.Replace(enumString)
+
+		if bicepType == "object" {
+			declareAllowedObject(cleanString, sch, buf)
+		} else {
+			buf.WriteString(fmt.Sprintf("@allowed(%v)\n", cleanString))
+		}
+	}
+
+	return nil
+}
+
+func declareAllowedObject(str string, sch *schema.Schema, buf *bytes.Buffer) {
+	splitString := strings.Split(str, " ")
+	joinList := []string{}
+	for _, d := range splitString {
+		if strings.Contains(d, ":") {
+			d = strings.ReplaceAll(d, `'`, "")
+		}
+		joinList = append(joinList, d)
+	}
+	bicepObj := strings.Join(joinList, " ")
+	buf.WriteString(fmt.Sprintf("@allowed(%v)\n", bicepObj))
+}
+
+func declareDescription(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Description != "" {
+		// decorators are in sys namespace. to avoid potential collision with other parameters named "description", we use "sys.description" instead of just "description" https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#decorators
+		buf.WriteString(fmt.Sprintf("@sys.description('%s')\n", sch.Description))
+	}
+}
+
+func declareMinValue(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Minimum != "" {
+		buf.WriteString(fmt.Sprintf("@minValue(%v)\n", sch.Minimum))
+	}
+}
+
+func declareMaxValue(sch *schema.Schema, buf *bytes.Buffer) {
+	if sch.Maximum != "" {
+		buf.WriteString(fmt.Sprintf("@maxValue(%v)\n", sch.Maximum))
+	}
+}
+
+func declareMinLength(sch *schema.Schema, buf *bytes.Buffer) error {
+	bicepType, err := getBicepType(sch.Type)
+	if err != nil {
+		return err
+	}
+
+	if bicepType == "array" {
+		if sch.MinItems != nil {
+			buf.WriteString(fmt.Sprintf("@minLength(%d)\n", *sch.MinItems))
+		}
+		return nil
+	}
+
+	if sch.MinLength != nil {
```

Comment:
This is only valid if the type is a `string`. We also perform this check on line 152, how can this function be restructured so that we don't call the same `if` twice?